tests/regression_tests/surface_source_write/test.py

```diff
@@ -0,0 +1,991 @@
+    folder, model_name, parameter, single_thread, single_process, request
+):
+    """Test on history-based results for CSG-only geometries."""
+    assert os.environ["OMP_NUM_THREADS"] == "1"
+    assert config["mpi_np"] == "1"
+    model = request.getfixturevalue(model_name)
+    model.settings.surf_source_write = parameter
+    harness = SurfaceSourceWriteTestHarness(
+        "statepoint.5.h5", model=model, workdir=folder
+    )
+    harness.main()
+
+
+@pytest.mark.skipif(config["event"] is True, reason="Results from history-based mode.")
+def test_consistency_low_realization_number(model_1, two_threads, single_process):
+    """The objective is to test that the results produced, in a case where
+    the number of potential realization (particle storage) is low
+    compared to the capacity of storage, are still consistent.
+
+    This configuration ensures that the competition between threads does not
+    occur and that the content of the source file created can be compared.
+
+    """
+    assert os.environ["OMP_NUM_THREADS"] == "2"
+    assert config["mpi_np"] == "1"
+    model_1.settings.surf_source_write = {
+        "max_particles": 200,
+        "surface_ids": [2],
+        "cellfrom": 2,
+    }
+    harness = SurfaceSourceWriteTestHarness(
+        "statepoint.5.h5", model=model_1, workdir="case-20"
+    )
+    harness.main()
+
+
+@pytest.mark.skipif(config["event"] is False, reason="Results from event-based mode.")
+@pytest.mark.parametrize(
+    "folder, model_name, parameter",
+    [
+        (
+            "case-21",
+            "model_1",
+            {"max_particles": 3000, "surface_ids": [4, 5, 6, 7, 8, 9], "cell": 2},
+        ),
+        ("case-22", "model_1", {"max_particles": 3000, "cell": 3}),
+        (
+            "case-23",
+            "model_2",
+            {"max_particles": 3000, "surface_ids": [4, 5, 6, 7, 8, 9], "cell": 3},
+        ),
+    ],
+)
+def test_surface_source_cell_event_based(
+    folder, model_name, parameter, single_thread, single_process, request
+):
+    """Test on event-based results for CSG-only geometries."""
+    assert os.environ["OMP_NUM_THREADS"] == "1"
+    assert config["mpi_np"] == "1"
+    model = request.getfixturevalue(model_name)
+    model.settings.surf_source_write = parameter
+    harness = SurfaceSourceWriteTestHarness(
+        "statepoint.5.h5", model=model, workdir=folder
+    )
+    harness.main()
+
+
+@pytest.fixture(scope="module")
+def model_dagmc_1():
+    """Model based on the mesh file 'dagmc.h5m' available from
+    tests/regression_tests/dagmc/legacy.
+
+    """
+    openmc.reset_auto_ids()
+    model = openmc.Model()
+
+    # =============================================================================
+    # Materials
+    # =============================================================================
+
+    u235 = openmc.Material(name="no-void fuel")
+    u235.add_nuclide("U235", 1.0, "ao")
+    u235.set_density("g/cc", 11)
+    u235.id = 40
```

Comment:
Do we really need to set the ID manually or can this be removed? (same thing below for water)

Reply:
The materials ID are required to run with the DAGMC model I am using in this case.